DependencyInjection/HttplugExtension.php

```diff
@@ -33,5 +36,31 @@ public function load(array $configs, ContainerBuilder $container)
         foreach ($config['main_alias'] as $type => $id) {
             $container->setAlias(sprintf('httplug.%s', $type), $id);
         }
+
+        // Configure services
+        foreach ($config['clients'] as $name => $arguments) {
+            $this->verifyDependencies($arguments['adapter']);
+            $def = $container->register('httplug.'.$name, DummyFactory::class);
+            $def->setFactory([new Reference('httplug.client.factory'), 'getClient'])
```

Comment:
I we use the specific factory direct we need to keep a list that maps name=>factory class. Because at this point we do not know any of the DI tags. That would mean if a user would like to add a factory e need to update/extend this bundle. 

Using a generic factory is mor SOLID.